shell/platform/windows/flutter_windows_view_unittests.cc

```diff
@@ -426,6 +426,117 @@ TEST(FlutterWindowsView, AddSemanticsNodeUpdateWithChildren) {
   }
 }
 
+// Flutter used to assume that the accessibility root had ID 0.
+// In a multi-view world, each view has its own accessibility root
+// with a globally unique node ID.
+//
+//        node1
+//          |
+//        node0
+//
+// node1 is a grouping node, node0 is a static text node.
+TEST(FlutterWindowsView, NonZeroSemanticsRoot) {
+  std::unique_ptr<FlutterWindowsEngine> engine = GetTestEngine();
+  EngineModifier modifier(engine.get());
+  modifier.embedder_api().UpdateSemanticsEnabled =
+      [](FLUTTER_API_SYMBOL(FlutterEngine) engine, bool enabled) {
+        return kSuccess;
+      };
+
+  auto window_binding_handler =
+      std::make_unique<::testing::NiceMock<MockWindowBindingHandler>>();
+  FlutterWindowsView view(std::move(window_binding_handler));
+  view.SetEngine(std::move(engine));
+
+  // Enable semantics to instantiate accessibility bridge.
+  view.OnUpdateSemanticsEnabled(true);
+
+  auto bridge = view.GetEngine()->accessibility_bridge().lock();
+  ASSERT_TRUE(bridge);
+
+  // Add root node.
+  FlutterSemanticsNode node1{sizeof(FlutterSemanticsNode), 1};
+  std::vector<int32_t> node1_children{0};
+  node1.child_count = node1_children.size();
+  node1.children_in_traversal_order = node1_children.data();
+  node1.children_in_hit_test_order = node1_children.data();
+
+  FlutterSemanticsNode node0{sizeof(FlutterSemanticsNode), 0};
+  node0.label = "prefecture";
+  node0.value = "Kyoto";
+
+  bridge->AddFlutterSemanticsNodeUpdate(&node1);
+  bridge->AddFlutterSemanticsNodeUpdate(&node0);
+  bridge->CommitUpdates();
+
+  // Look up the root windows node delegate.
+  auto node_delegate = bridge->GetFlutterPlatformNodeDelegateFromID(1).lock();
```

Comment:
Should we also test that `GetFlutterPlatformNodeDelegateFromID(0)` returns `node0` as we would expect? If the rest of the test passes, I'd think it would, but it may still be useful to make sure arbitrary ID ordering works.

Reply:
Good idea. I made the root ID `1` and the child ID `2` and updated the tests to:

1. Verify node ID `0` is unused
2. Verify node ID `1` and `2` exist as expected